src/platform/packages/private/kbn-generate-csv/src/generate_csv.ts

```diff
@@ -426,6 +428,15 @@ export class CsvGenerator {
 
         // update iterator
         currentRecord += table.rows.length;
+
+        // stop generating the report if the
+        // current number of rows is >= the max row limit
+        if (currentRecord >= maxRows - 1) {
+          this.logger.warn(
+            `This search has exceeded the recommended row limit (${maxRows}). This limit can be configured in kibana.yml, but increasing it may impact performance.`
+          );
+          break;
```

Comment:
When this happens I see this in the report warnings:

<img width="372" height="263" alt="Screenshot 2025-11-25 at 10 38 52 AM" src="https://github.com/user-attachments/assets/1a366b7d-17ee-4713-bef0-fe1ec16433c8" />

Seems like we shouldn't have that second warning (since we short circuited intentionally) and instead have the max rows warning?

Reply:
I updated the messages here, https://github.com/elastic/kibana/pull/242603/commits/d6f267617261d9641e542b2829c981d427a7744f. cc @nastasha-solomon 

I didn't update the `kibana.yml` part yet, let me know if you'd like me to go ahead and change that too.

Reply:
Updated in this commit, [44d1df7](https://github.com/elastic/kibana/pull/242603/commits/44d1df79eed7b085f598de03466a71c4a2bc1cc7)